aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -58,13 +58,66 @@ static inline Tensor& unary_op_impl_(Tensor& self, OutImpl& out_impl) {
   return out_impl(self, self);
 }
 
-Tensor& acos_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, acos_stub); }
-Tensor acos(const Tensor& self) { return unary_op_impl(self, at::acos_out); }
-Tensor& acos_(Tensor& self) { return unary_op_impl_(self, at::acos_out); }
+// unary_floating_ufunc_* functions are helper functions to support integral and bool Tensors
+// to default floating type promotion for floating ufuncs. For non-floating ufuncs, use existing
+// helper functions (unary_op_impl*) if needed.
+template <typename Stub>
+static inline Tensor& unary_floating_ufunc_op_impl_out(Tensor& result, const Tensor& self, Stub& stub) {
+  TORCH_CHECK(!isIntegralType(result.scalar_type(), /*includeBool=*/ true),
+    "Attempted to call a function that returns a tensor of floating dtype ",
+    "but requested a tensor of bool or integral dtype. ");
+
+  auto iter = TensorIterator::unary_floating_ufunc(result, self,
+    /*check_mem_overlap=*/true);
+  stub(iter.device_type(), iter);
+  return result;
+}
+
+template <typename OutImpl>
+static inline Tensor unary_floating_ufunc_op_impl(const Tensor& self, OutImpl& out_impl) {
+  if (isIntegralType(self.scalar_type(), /*includeBool=*/ true)) {
+    const auto scalar_type = typeMetaToScalarType(c10::get_default_dtype());
+    Tensor result = at::empty({0}, self.options().dtype(scalar_type));
```

Comment:
I think you can just use `{}` here to avoid allocating space for the 0.

Reply:
not sure what you mean by allocating space for the 0, but `{}` would allocate a 1-element tensor.